scipy/optimize/tests/test_minpack.py

```diff
@@ -551,6 +551,75 @@ def test_NaN_handling(self):
         assert_raises(ValueError, curve_fit, lambda x, a, b: a*x + b,
                       xdata, ydata, **{"check_finite": True})
 
+    @staticmethod
+    def _check_nan_policy(f, xdata_with_nan, xdata_without_nan,
+                          ydata_with_nan, ydata_without_nan, method):
+        # propagate test
```

Comment:
It takes some time to see what arguments are actually changed in each call to `curve_fit`. Consider consolidating common values into `kwargs`. 
```suggestion
        kwargs = {'f': f, 'xdata': xdata_with_nan, 'ydata': ydata_with_nan,
                  'method': method, 'check_finite': False}
        # propagate test
```
That way, it's easier to tell what is really changing in each test. It's also shorter and easier to satisfy PEP8.